content/documentation/pages/5-feature-guides/3-batch/7-java-dsl.md

```diff
@@ -0,0 +1,163 @@
+
+<!--NOTE-->
+
+In order to create and launch your tasks, you need to make sure that the corresponding apps have been registered in the Data Flow server first.
+Attempting to launch a task that contains an unknown application throws an exception.
+You can register your application by using the `DataFlowTemplate`, as follows:
+
+```java
+dataFlowOperations.appRegistryOperations().importFromResource(
+            "https://dataflow.spring.io/task-maven-latest", true);
+```
+
+<!--END_NOTE-->
+
+Instead of creating a new Task you can use the `TaskBuilder` to retrieve an existing Task instances by name:
+
+```Java
+Optional<Task> task = Task.builder(dataflowOperations).findByName("myTask");
+```
+
+or to list all existing tasks:
+
+```Java
+List<Task> tasks = Task.builder(dataflowOperations).allTasks();
+```
+
+With the `Task` instance, you have methods available to `launch` or `destroy` the task.
+The following example launches the task:
+
+```Java
+long launchId = task.launch();
+
+```
+
+The `launchId` is an unique Task execution identifier for the launched task.
+The `launch` method is overloaded to take a `java.util.Map<String, String>` of launch properties and `java.util.List<String>` of the command line arguments.
+
+The tasks are asynchronously executed. If your use case require to wait on task completion or other task state you can you the java concurrency utils or the `Awaitility` library like this:
+
+```Java
+org.awaitility.Awaitility.await().until(
+  () -> task.executionStatus(launchId) == TaskExecutionStatus.COMPLETE);
+```
+
+The `Task` instance provides `executionStatus`, `destroy`, and `stop` methods to control and query the task.
+
+The `Collection<TaskExecutionResource> executions()` method list all `TaskExecutionResource`s launched by the task. Use the `launchId` to retrieve the `TaskExecutionResource` for a specific execution (`Optional<TaskExecutionResource> execution(long executionId)`).
+
+Similarly the `Collection<JobExecutionResource> jobExecutionResources()` and `Collection<JobInstanceResource> jobInstanceResources()` would let you introspect any Spring Batch Jobs when the task uses such.
+
+## TaskSchedule DSL Usage
+
+Consider the following example, which creates a new task and schedule it:
+
+```Java
+Task task = Task.builder(dataflowOperations)
+              .name("myTask")
+              .definition("timestamp")
+              .description("simple task")
+              .create();
+
+TaskSchedule schedule = TaskSchedule.builder(dataFlowOperations)
+              .prefix("mySchedule")
+              .task(task)
+              .create();
+```
+
+The `TaskScheduleBuilder` class is returned from the `TaskSchedule.builder(dataFlowOperations)` method.
+
+The `create` method returns an instance of a `TaskSchedule` instance called `mySchedule` configured whit a task instance.
+At this point the schedule is not yet scheduled to run launch the task.
+
+Use the schedule() method to trigger the schedule process:
```

Comment:
Use the schedule() method to create the schedule:

Reply:
Same can we use start instead of create here?

Reply:
If not then perhaps we should rename the builder's `create()` method into `build()` to avoid confusion?

Reply:
Ok i've renamed the `TaskScheduleBuilder#create()` into `TaskScheduleBuilder#build()`and now can use the suggested description. 
For consistency i've also renamed `TaskBuilder#create()` into `TaskBuilder#build()`